scripts/BackupRetentionScript/RetentionScript.ps1

```diff
@@ -0,0 +1,194 @@
+    example:
+    ManagementEndpoint : https://clustername.centralus.cloudapp.azure.com:19080
+    $ClusterEndPoint : clustername.centralus.cloudapp.azure.com:19080
+
+ .PARAMETER DeleteNotFoundPartitions(Optional)
+    If there are partitions on the storage which are not found on the cluster, and you want to delete complete data in the partition
+    then, enable the flag and run the script.
+    
+ .PARAMETER PartitionId(Optional)
+    Filter to delete data for a particular partition on the cluster
+        
+ .PARAMETER ApplicationId
+    Filter to delete data for a particular Application on the cluster
+    Example:
+    if ApplicationName(without fabric:) is  "application/Name" then,
+    $ApplicationId = "application~Name"(Replace "/" with "~")
+        
+ .PARAMETER ServiceId
+    Filter to delete data for a particular service on the cluster
+    Example:
+    if ServiceName(without fabric:) is  "Service/Name" then,
+    $ServiceId = "Service~Name"(Replace "/" with "~")
+        
+ .PARAMETER ClientCertificateThumbprint(Required in case of secured cluster)
+    Thumbprint of the client certificate
+#>
+
+[CmdletBinding(PositionalBinding = $false)]
+param (
+    [Parameter(Mandatory=$false)]
+    [String] $UserName,
+
+    [Parameter(Mandatory=$false)]
+    [String] $FileSharePath,
+
+    [Parameter(Mandatory=$true)]
+    [String] $StorageType,
+
+    [Parameter(Mandatory=$true)]
+    [String] $DateTimeBefore,
+    
+    [Parameter(Mandatory=$false)]
+    [String] $ConnectionString,
+    
+    [Parameter(Mandatory=$false)]
+    [SecureString] $Password,
+
+    [Parameter(Mandatory=$false)]
+    [String] $ContainerName,
+
+    [Parameter(Mandatory=$false)]
+    [string] $StorageAccountName,
+
+    [Parameter(Mandatory=$false)]
+    [String] $StorageAccountKey,
+
+    [Parameter(Mandatory=$true)]
+    [String] $ClusterEndPoint,
+
+    [Parameter(Mandatory=$false)]
+    [Switch] $DeleteNotFoundPartitions,
+
+    [Parameter(Mandatory=$false)]
+    [String] $PartitionId,
+
+    [Parameter(Mandatory=$false)]
+    [String] $ServiceId,
+ 
+    [Parameter(Mandatory=$false)]
+    [String] $ApplicationId,
+
+    [Parameter(Mandatory=$false)]
+    [String] $ClientCertificateThumbprint
+)
+
+$command = ""
+if($StorageType -eq "FileShare")
+{
+  if(!$FileSharePath)
+  {
+    $FileSharePath = Read-Host -Prompt "Please enter the FileShare path"
+  }
+
+  if($UserName)
+  {
+      $command = $command +  ".\RetentionScriptFileShare.ps1 -UserName `"$UserName`" -FileSharePath `"$FileSharePath`" -DateTimeBefore `"$DateTimeBefore`" -ClusterEndPoint `"$ClusterEndPoint`""
+    if(!$Password)
+    {
+        $Password = Read-Host -Prompt "Please enter password for the userName: $UserName" -AsSecureString
+    }
+    $Global:Pass = $Password
+  }
+  else {
+    $command = $command +  ".\RetentionScriptFileShare.ps1 -FileSharePath `"$FileSharePath`" -DateTimeBefore `"$DateTimeBefore`" -ClusterEndPoint `"$ClusterEndPoint`""
```

Comment:
>".\RetentionScriptFileShare.ps1 [](start = 27, length = 31)

We should include module and directly call respective function with parameters instead of calling another script ...

Reply:
not fixing it.